docs/csharp/whats-new/csharp-12.md

```diff
@@ -21,6 +23,64 @@ The compiler generates public properties for primary constructor parameters only
 
 You can learn more about primary constructors in the tutorial for [exploring primary constructors](./tutorials/primary-constructors.md) and in the article on [instance constructors](../programming-guide/classes-and-structs/instance-constructors.md#primary-constructors).
 
+## Collection expressions
+
+Collection expressions introduce a new terse syntax, to create common collection values. Inlining other collections into these values is possible using a spread operator `..`.
+
+Several collection-like types can be created without requiring external BCL support.  These types are:
+
+* Array types, such as `int[]`.
+* <xref:System.Span%601?displayProperty=nameWithType> and <xref:System.ReadOnlySpan%601?displayProperty=nameWithType>.
+* Types that support collection initializers, such as <xref:System.Collections.Generic.List%601?displayProperty=nameWithType>.
+
+The following examples show uses of collection expressions:
+
+```csharp
+// Create an array:
+int[] a = [1, 2, 3, 4, 5, 6, 7, 8];
+
+// Create a span
+Span<int> b  = ['a', 'b', 'c', 'd', 'e', 'f', 'h', 'i'];
+
+// Create a 2 D array:
+int[][] twoD = [[1, 2, 3], [4, 5, 6], [7, 8, 9]];
+
+// create a 2 D array from variables:
+int[] row0 = [1, 2, 3];
+int[] row1 = [4, 5, 6];
+int[] row2 = [7, 8, 9];
+int[][] twoDFromVariables = [row0, row1, row2];
+
+```
+
+The *spread operator*, `..` in a collection expression replaces its argument with the elements from that collection. The argument must be a collection type. The following examples show how the spread operator works:
+
+```csharp
+int[] row0 = [1, 2, 3];
+int[] row1 = [4, 5, 6];
+int[] row2 = [7, 8, 9];
+int[] single = [..row0, ..row1, ..row2];
+foreach (var element in single)
+{
+    Console.Write($"{element}, ");
+}
+// output:
+// 1, 2, 3, 4, 5, 6, 7, 8, 9,
+
+int[] vector = [..a[0..^2]];
```

Comment:
The preceding example is easy to follow, but this one I have trouble parsing. You could consider adding some text explaining it. Or omit it if the preceding example adequately illustrates how the spread operator works.

Reply:
I removed that second sample. I'll elaborate on it in the language reference updates, coming in the next PR.